metamorph/processor.go

```diff
@@ -209,6 +214,17 @@ func (p *Processor) Shutdown() {
 	}
 }
 
+func (p *Processor) unlockItems() error {
+	items := p.processorResponseMap.Items()
+	hashes := make([]*chainhash.Hash, len(items))
+	index := 0
+	for key := range items {
+		hashes[index] = &key
```

Comment:
you are setting hashes[0] with new keys throughout the loop here. 
probably you meant 
```
for index, key := range items {
  hashes[index] = &key
}
``` 
?)

Reply:
This I can't do because it is a map, so the first parameter is not an index.

But you're right, I forgot thi `index++`